src/lib/hooks/useCurrencyBalance.ts

```diff
@@ -72,11 +72,15 @@ export function useTokenBalancesWithLoadingIndicator(
 
   const anyLoading: boolean = useMemo(() => balances.some((callState) => callState.loading), [balances])
 
-  return useMemo(
-    () => [
+  // this is a hack to get around the fact that multicall doesn't return the same array reference when it updates
+  // TODO: move this to a MOD file
+  const cachedBalances = useMemo(() => balances, [address, validatedTokens, anyLoading])
```

Comment:
There's a huge difference in performance once you connect the wallet and the balances update starts triggering a cascade of renderings throughout the components. This can be observed by using the Select a token modal and the Tokens table with and without being connected to a wallet